hnn_core/network_builder.py

```diff
@@ -276,6 +276,9 @@ def __init__(self, net):
         # initialized by _ArtificialCell()
         self._feed_cells = []
         self.ncs = dict()
+
+        self.net._create_gid_dict()
```

Comment:
Is there really any (logial/semantic) reason for GID creation to occur before `NetworkBuilder`? This would allow a `Network` to be constructed incrementally by specifying the real cell types, the spatial grid, and finally the `Drive`s ("feeds"). GIDs are only truly needed when cells need to be connected.

Reply:
I think the idea was to separate the `Neuron` code from the rest because Neuron code is not picklable. But I've come around to the idea that they can reside in the same class as long as the Neuron code is called within specific methods

Reply:
I decided to leave the storage of the GIDs into `Network`, just moved initialisation to `NetworkBuilder` where it belongs.

Reply:
I don't quite get this ... it seems very convoluted to me. Why initialize an attribute of one class from another class? Folks will have a tough time following the code in the future ...